softlayer/vm/softlayer_vm.go

```diff
@@ -135,6 +150,21 @@ func (vm SoftLayerVM) AttachDisk(disk bslcdisk.Disk) error {
 		return bosherr.WrapErrorf(err, "Failed to attach volume with id %d to virtual guest with id: %d.", volume.Id, virtualGuest.Id)
 	}
 
+	metadata, err := bslcommon.GetUserMetadataOnVirtualGuest(vm.softLayerClient, virtualGuest.Id)
+
+	old_agentEnv, err := NewAgentEnvFromJSON(metadata)
+	new_agentEnv := old_agentEnv.AttachPersistentDisk(strconv.Itoa(disk.ID()), "/dev/sda")
```

Comment:
Please add a comment to say that this is always the case in SL and any links to documentation if you have.